File: 1000/1567B/main.cpp
#include<bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int a, b;
        cin >> a >> b;

        int xor_val = 0;
        int n = a - 1;

        // Calculate the XOR value for the range
        // XOR prefix sum from 0 to a-1
        if (n % 4 == 0) 
            xor_val = n;
        else if (n % 4 == 1) 
            xor_val = 1;
        else if (n % 4 == 2) 
            xor_val = n + 1;
        else if (n % 4 == 3) 
            xor_val = 0;

        // Determine the output based on the XOR values
        if(xor_val == b)
            cout << a << "\n";
        else if((xor_val ^ a) == b)
            cout << a + 2 << "\n";
        else
            cout << a + 1 << "\n";
    }

    return 0;
}

File: 1000/1567B/main.java
import java.util.*;
public class main 
{
    public static void main(String args[]) 
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
 
        while(t-- > 0)
        {
            int a = sc.nextInt();
            int b = sc.nextInt();
 
            int xor = 0;

            // Calculate the XOR value for the range
            // XOR prefix sum from 0 to a-1
            if ((a - 1) % 4 == 0) 
                xor = a - 1;
            else if ((a - 1) % 4 == 1) 
                xor = 1;
            else if ((a - 1) % 4 == 2) 
                xor = a;

            // Determine the output based on the XOR values
            if (xor == b) 
                System.out.println(a);
            else if ((xor ^ a) == b) 
                System.out.println(a + 2);
            else 
                System.out.println(a + 1);
        }
    }
}
